hal-core/src/boot.rs

```diff
@@ -4,10 +4,14 @@ use core::iter::Iterator;
 pub trait BootInfo {
     type Arch: Architecture;
     type MemoryMap: Iterator<Item = mem::Region<Self::Arch>>;
+    type Writer: core::fmt::Write;
```

Comment:
We might want to introduce our own wrapper `Write` trait, like `std::io::Write`, which handles proper error propagation. Perhaps in the util library?
It re-wraps formatters to bubble errors out: https://github.com/rust-lang/rust/blob/3964a55ba59bb6c3d4badcbddc49f6929ef76862/src/libstd/io/mod.rs#L1404-L1437

Reply:
Yeah, I think this is probably a good idea — I want to do this in a follow-up

Reply:
Opened  #29